nibabel/wrapstruct.py

```diff
@@ -536,8 +536,18 @@ def __str__(self):
 
         def _getter(obj, key):
             try:
-                return obj.get_value_label(key)
+                val = obj.get_value_label(key)
             except ValueError:
-                return obj[key]
+                val = obj[key]
+            # Convert a dtype('S1') numpy string to a Python str, if possible
+            if isinstance(val, np.ndarray) and val.dtype.type == np.bytes_:
+                try:
+                    tmp = val.tolist().decode()
+                    # Do not return unicode objects (Python 2)
+                    if isinstance(tmp, str):
```

Comment:
I don't understand what's going on here--why there's an error in decoding in Python 2, and why this code is necessary for Python 3.

Assuming this was working in Python 2, and is now working for Python 3, seems there should be a way to get it working for both. For example, is it not possible to use `val = str(val)` in Python 2, if the decode fails? That may be nonsense for me to suggest; I'm not too clear on the issue this code is trying to solve.